File: SocialAppFlutter/social_app/lib/services/events/create_event_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'json_generated_code/create_event_request.g.dart';

@JsonSerializable(explicitToJson: true)
class CreateEventRequest {
  final int sportCategoryId;
  final int locationId;
  final int creatorId;
  final int requiredMembers;
  final List<int> allMembers;
  final DateTime startDatetime;
  final double duration;
  final bool createNewLocation;
  double lat = 0.0;
  double long = 0.0;
  String city = '';
  String locationName = '';

  CreateEventRequest(
      this.sportCategoryId,
      this.locationId,
      this.creatorId,
      this.requiredMembers,
      this.startDatetime,
      this.duration,
      this.allMembers,
      this.createNewLocation,
      this.lat,
      this.long,
      this.city,
      this.locationName);

  Map<String, dynamic> toJson() => _$CreateEventRequestToJson(this);
}

File: APIs/EventsService/EventsService/DTOs/Request/CreateEventRequest.cs
namespace EventsService.DTOs.Request
{
    public class CreateEventRequest
    {
        public int SportCategoryId { get; set; }
        public int LocationId { get; set; }
        public int CreatorId { get; set; }
        public int RequiredMembers { get; set; }
        public List<int> AllMembers { get; set; } = null!;
        public DateTime StartDatetime { get; set; }
        public double Duration { get; set; }
        public bool CreateNewLocation { get; set; }
        public double Lat { get; set; }
        public double Long { get; set; }
        public string City { get; set; } = string.Empty;
        public string LocationName { get; set; } = string.Empty;
    }
}
